rmgpy/rmg/input.py

```diff
@@ -1510,25 +1515,73 @@ def save_input_file(path, rmg):
+            f.write('surfaceReactor(\n')
+            # determine if it is a ranged reactor
+            if system.T:
+                f.write('    temperature = ({0:g}, {1!s}),\n'.format(system.T.value, system.T.units))
+            else: 
+                Trange_lo = (system.Trange[0].value, system.Trange[0].units)
+                Trange_hi = (system.Trange[1].value, system.Trange[1].units)
+                f.write('    temperature = [{0}, {1}],\n'.format(Trange_lo, Trange_hi))
+
+            # Convert the pressure from SI pascal units to bar here
+            # Do something more fancy later for converting to user's desired units for both T and P..
+            if system.P_initial:
+                f.write('    initialPressure = ({0:g}, "{1!s}"),\n'.format(system.P_initial.value, system.P_initial.units))
+            else:
+                Prange_lo = (system.Prange[0].value, system.Prange[0].units)
+                Prange_hi = (system.Prange[1].value, system.Prange[1].units)
+                f.write('    temperature = [{0}, {1}],\n'.format(Prange_lo, Prange_hi))
+
+            f.write('    initialGasMoleFractions={\n')
+            for spcs, molfrac in system.initial_gas_mole_fractions.items():
+                f.write('        "{0!s}": {1:g},\n'.format(spcs.label, molfrac))
+            f.write('    },\n')
+
+            f.write('    initialSurfaceCoverages={\n')
+            for spcs, molfrac in system.initial_surface_coverages.items():
+                f.write('        "{0!s}": {1:g},\n'.format(spcs.label, molfrac))
+            f.write('    },\n')
+
+            f.write('    surfaceVolumeRatio=({0:g},"{1!s}"),\n'.format(system.surface_volume_ratio.value, system.surface_volume_ratio.units))
+
         else:
             f.write('simpleReactor(\n')
-            f.write('    temperature = ({0:g},"{1!s}"),\n'.format(system.T.value, system.T.units))
+            if system.T:
+                f.write('    temperature = ({0:g}, {1!s}),\n'.format(system.T.value, system.T.units))
+            else: 
+                Trange_lo = (system.Trange[0].value, system.Trange[0].units)
+                Trange_hi = (system.Trange[1].value, system.Trange[1].units)
+                f.write('    temperature = [{0}, {1}],\n'.format(Trange_lo, Trange_hi))
+
             # Convert the pressure from SI pascal units to bar here
             # Do something more fancy later for converting to user's desired units for both T and P..
-            f.write('    pressure = ({0:g},"{1!s}"),\n'.format(system.P.value, system.P.units))
+            if system.P_initial:
+                f.write('    pressure = ({0:g}, {1!s}),\n'.format(system.P_initial.value, system.P_initial.units))
+            else:
+                Prange_lo = (system.Prange[0].value, system.Prange[0].units)
+                Prange_hi = (system.Prange[1].value, system.Prange[1].units)
+                f.write('    temperature = [{0}, {1}],\n'.format(Prange_lo, Prange_hi))
```

Comment:
Why is `temperature` being set to something called `Prange`? if this is correct, then the variable names are confusing and likely to cause future errors.